kernel/src/table_changes/physical_to_logical.rs

```diff
@@ -79,62 +44,361 @@ pub(crate) fn scan_file_physical_schema(
+                transform.with_inserted_field(insert_after.clone(), expr.clone())
+            }
+            StaticReplace { field_name, expr } => {
+                transform.with_replaced_field(field_name.clone(), expr.clone())
+            }
+            StaticDrop { field_name } => transform.with_dropped_field(field_name.clone()),
+            PartitionColumn {
+                field_index,
+                insert_after,
+            } => {
+                if let Some((_, field_value)) = partition_field_values.get(field_index) {
+                    let field_value = Arc::new(field_value.clone());
+                    let actual_insert_after = resolve_insert_after_with_change_type_redirection(
+                        insert_after.clone(),
+                        change_type_insert_after.clone(),
+                    );
+                    transform.with_inserted_field(actual_insert_after, field_value)
+                } else {
+                    return Err(crate::Error::InternalError(format!(
+                        "Missing field value for partition column at index {field_index}"
+                    )));
+                }
+            }
+            Cdf {
+                field_index,
+                insert_after,
+                col_type: _,
+            } => {
+                let actual_insert_after = resolve_insert_after_with_change_type_redirection(
+                    insert_after.clone(),
+                    change_type_insert_after.clone(),
+                );
+
+                if let Some((_, field_value)) = cdf_field_values.get(field_index) {
+                    let field_value = Arc::new(field_value.clone());
+                    transform.with_inserted_field(actual_insert_after, field_value)
+                } else {
+                    // CDF field was not generated, meaning it should come from physical data
+                    // (e.g., ChangeType column in CDC files)
+                    transform
+                }
+            }
+        }
+    }
+
+    Ok(Arc::new(Expression::Transform(transform)))
+}
+
+/// Resolves the correct insert_after field for a column, applying change_type redirection when needed.
+///
+/// When a column wants to insert after "_change_type" but the change type column itself
+/// needs to be redirected (e.g., from "_change_type" to "name"), this function handles
+/// the redirection so the column goes to the correct location.
+///
+/// Example: CommitVersion wants to go after "_change_type", but "_change_type" goes after "name"
+/// → CommitVersion gets redirected to go after "name" instead
+fn resolve_insert_after_with_change_type_redirection(
+    insert_after: Option<String>,
+    change_type_insert_after: Option<String>,
+) -> Option<String> {
+    if insert_after
+        .as_ref()
+        .map(|field| field == CHANGE_TYPE_COL_NAME)
+        .unwrap_or(false)
+    {
+        change_type_insert_after.or(insert_after)
+    } else {
+        insert_after
+    }
+}
```

Comment:
Let's also take references so we don't have to clone both `insert_after` _and_ `change_type_insert_after` if they are both `Some`


```suggestion
fn resolve_insert_after_with_change_type_redirection(
    insert_after: &Option<String>,
    change_type_insert_after: &Option<String>,
) -> Option<String> {
    if insert_after.as_ref().is_some_and(|field| field == CHANGE_TYPE_COL_NAME) {
        change_type_insert_after.clone().or_else(|| {
            insert_after.clone()
        })
    } else {
        insert_after.clone()
    }
}
```

And then above you can do:

```rust
let actual_insert_after = resolve_insert_after_with_change_type_redirection(
  &insert_after,
  &change_type_insert_after,
);